client/checkout/woopay/express-button/woopay-express-checkout-button.js

```diff
@@ -248,14 +248,14 @@ export const WoopayExpressCheckoutButton = ( {
 								'woocommerce-payments'
 							);
 							showErrorMessage( context, errorMessage );
-							isLoadingRef.current = false;
+							setIsLoading( false );
 						} );
 				}
 			};
 		} );
 
 		return iframe;
-	}, [ isProductPage, context, isPreview, listenForCartChanges ] );
+	}, [ isProductPage, isPreview, isLoading, context, listenForCartChanges ] );
```

Comment:
@ricardo, I just wanted to jump in here and highlight that by adding `isLoading` here, we inadvertently introduce a regression. For example, once we click the WooPay button, the `isLoading` state is updated to `true`, which in turn calls the `const newIframe = useCallback()`, which ultimately calls the `useEffect()` that appends the iFrame to the UI. This means we end up with duplicate iFrames in the UI. Once the `isLoading` state returns to `false` (for any given reason), we add another iFrame to the UI.

I suggest maintaining the `isLoadingRef` alongside `isLoading` and removing the `isLoading` dependency from `const newIframe = useCallback()`. Although not the ideal solution, this will prevent a `newIframe` from being created and added to the UI each time `isLoading` state updates. We may also want to add a note about this potential regression next to the dependencies.

Reply:
Thanks for pointing this out. I didn't notice the duplicate iframe when testing.

Indeed it doesn't seem ideal to keep both the `isLoading` state and `isLoadingRef`, but I also don't have a better suggestion.

Fixed in https://github.com/Automattic/woocommerce-payments/pull/7366/commits/6b351fcdd48fe2f4f451a2e7c80d02246747c302.

@lovo-h Not suggesting any changes, but just wondering, could the iframe be rendered as a React component inside the button perhaps? To avoid these non-react DOM manipulations.

Reply:
> @lovo-h Not suggesting any changes, but just wondering, could the iframe be rendered as a React component inside the button perhaps? To avoid these non-react DOM manipulations.

In other places, we compose the iframe like we do here. But I think that if we render the iframe like a React component, that would address these issues. Good call.